backends/k8s/client.go

```diff
@@ -0,0 +1,219 @@
+package k8s
+
+import (
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"fmt"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/pkg/runtime/schema"
+	"k8s.io/client-go/pkg/runtime/serializer"
+	"k8s.io/client-go/pkg/runtime"
+	"github.com/kelseyhightower/confd/log"
+	"github.com/projectcalico/libcalico-go/lib/backend/k8s/thirdparty"
+	kapiv1 "k8s.io/client-go/pkg/api/v1"
+	metav1 "k8s.io/client-go/pkg/apis/meta/v1"
+	clientapi "k8s.io/client-go/pkg/api"
+	kerrors "k8s.io/client-go/pkg/api/errors"
+	"time"
+	"regexp"
+	"encoding/json"
+	"github.com/projectcalico/libcalico-go/lib/net"
+)
+
+const (
+	ipPool          = "/calico/v1/ipam/v4/pool"
+	global          = "/calico/bgp/v1/global"
+	globalASN       = "/calico/bgp/v1/global/as_num"
+	globalNodeMesh  = "/calico/bgp/v1/global/node_mesh"
+	allNodes        = "/calico/bgp/v1/host"
+	globalLogging   = "/calico/bgp/v1/global/loglevel"
+)
+
+var (
+	singleNode = regexp.MustCompile("^/calico/bgp/v1/host/([a-zA-Z0-9._/-]*)")
+)
+
+type Client struct {
+	clientSet *kubernetes.Clientset
+	tprClient *rest.RESTClient
+}
+
+type KubeConfig struct {
+	Kubeconfig     string
+	K8sAPIEndpoint string
+	K8sKeyFile     string
+	K8sCertFile    string
+	K8sCAFile      string
+	K8sAPIToken    string
+}
+
+func NewK8sClient(kubeconfig string) (*Client, error) {
+
+	kc := KubeConfig {
```

Comment:
Good point, I had removed all the CLI flags with host, port etc... This serves no point anymore.